File: Java/String1/HelloName.java
/*
 * CodingBat Java Activity "helloName" from String-1 section.
 * codingbat.com
 */

public class HelloName 
{
	private static int failures = 0;
	private static final String METHOD_NAME = "helloName";
	
	/*
	 * INSTRUCTIONS
	 * Given a string name, e.g. "Bob", return a greeting of the form "Hello Bob!".
	 * 
	 * helloName("Bob") -> "Hello Bob!"
	 * helloName("Alice") -> "Hello Alice!"
	 * helloName("X") -> "Hello X!"
	 */
	public static String helloName(String name)
	{
		//put your code ONLY here!
	}
	
	public static void main(String[] args)
	{
		testCase("\"Bob\"","\"Hello Bob!\"");
		testCase("\"Alice\"", "\"Hello Alice!\"");
		testCase("\"X\"", "\"Hello X!\"");
		testCase("\"Dolly\"", "\"Hello Dolly!\"");
		testCase("\"Alpha\"", "\"Hello Alpha!\"");
		testCase("\"Omega\"", "\"Hello Omega!\"");
		testCase("\"Goodbye\"", "\"Hello Goodbye!\"");
		testCase("\"ho ho ho\"", "\"Hello ho ho ho!\"");
		testCase("\"xyz!\"", "\"Hello xyz!!\"");
		testCase("\"Hello\"", "\"Hello Hello!\"");
		
		System.out.println();
		System.out.println(failures + " test cases failed.");
	}

	private static void testCase(String a, String result)
	{
		String r;
		try
		{
			r = ""+ "\"" + helloName(a.substring(1,a.length()-1)) + "\"";
			if (r.equals(""+result))
			{
				System.out.print("PASS: ");
			}
			else
			{
				System.out.println("FAIL: ");
				failures++;
			}
		}
		catch (Exception e)
		{
			r = e.getMessage();
			System.out.print("FAIL: ");
			failures++;
		}
		System.out.println(METHOD_NAME+"("+a+") -> "+r + ", EXPECTED: "+result);
	}

}

File: Python/string-1/hello_name.py
'''
CodingBat Python Activity "hello_name" from string-1.
codingbat.com
'''

failures = 0

def hello_name(name):
    '''
Given a string name, e.g. "Bob", return a greeting of the form "Hello Bob!".


hello_name('Bob') → 'Hello Bob!'
hello_name('Alice') → 'Hello Alice!'
hello_name('X') → 'Hello X!'
'''
    pass #Replace pass with your code to implement this function





def catchFailures(testCase, result):
    global failures
    
    try:
        assert eval(testCase) == result, "Test Case Failed: "+testCase+ " → "+str(result)
    except Exception as e:
        failures += 1
        print(e)
            


def test():
    global failures
    
    catchFailures("hello_name('Bob')", 'Hello Bob!')
    catchFailures("hello_name('Alice')", 'Hello Alice!')
    catchFailures("hello_name('X')", 'Hello X!')
    catchFailures("hello_name('Dolly')", 'Hello Dolly!')
    catchFailures("hello_name('Alpha')", 'Hello Alpha!')
    catchFailures("hello_name('Omega')", 'Hello Omega!')
    catchFailures("hello_name('Goodbye')", 'Hello Goodbye!')
    catchFailures("hello_name('ho ho ho')", 'Hello ho ho ho!')
    catchFailures("hello_name('xyz!')", 'Hello xyz!!')
    catchFailures("hello_name('Hello')", 'Hello Hello!')

        
    if failures > 0:
        print("\r\n "+ str(failures) + " failed test case(s)   :(")
    else:
        print("Good job -- problem solved!")


test()
